bee-ledger/src/workers/snapshot/config.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Module containing snapshot configuration.
+
+use serde::Deserialize;
+
+use std::path::{Path, PathBuf};
+
+const DEFAULT_FULL_PATH: &str = "./snapshots/mainnet/latest-full_snapshot.bin";
+const DEFAULT_DOWNLOAD_URLS: Vec<String> = Vec::new();
+const DEFAULT_DEPTH: u32 = 50;
+const DEFAULT_INTERVAL_SYNCED: u32 = 50;
+const DEFAULT_INTERVAL_UNSYNCED: u32 = 1000;
+
+/// Builder for a `SnapshotConfig`.
+#[derive(Default, Deserialize)]
+pub struct SnapshotConfigBuilder {
+    full_path: Option<String>,
```

Comment:
Perhaps consider just using `PathBuf` here? It's got tighter constraints than `String`.

Reply:
Using String for serde I guess

Reply:
![image](https://user-images.githubusercontent.com/3268430/117851883-a9d71280-b27e-11eb-9270-275939d4e1f4.png)

There's an implementation for `PathBuf` too.

Reply:
Ah, I missed that, perfect then